File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n = numbers.length;
        for(int i = 1; i <= n; i++) {
            int theOne = target - numbers[i-1];
            for(int j = i + 1; j <= n; j++) {
                if(numbers[j-1] == theOne) {
                    return new int[]{i, j};
                }
            }
        }
        return new int[]{0, 0};
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.go
func twoSum(nums []int, target int) []int {
	hash := make(map[int]int)
	result := make([]int, 2)
	for i, v := range nums {
		hash[v] = i
	}
	fmt.Println(hash)
	for i, v := range nums {
		difference := target - v
		value, ok := hash[difference]
		if ok && value != i {
			result[0] = i
			result[1] = value
			return result
		} else {
			continue
		}
	}
	return result

}